androidApp/src/main/java/com/example/tasky/android/agenda/screen/AgendaDetailsScreen.kt

```diff
@@ -50,6 +63,55 @@ enum class AgendaDetailsScreenType {
     REMINDER,
 }
 
+fun NavGraphBuilder.agendaDetailsScreen(navigateUp: () -> Unit) {
+    composable<AgendaDetails>(typeMap = AgendaDetails.typeMap) {
+        val args = it.toRoute<AgendaDetails>()
+        val viewModel: AgendaDetailsViewModel =
+            koinViewModel {
+                parametersOf(args.agendaId, args.type, args.isEdit)
+            }
```

Comment:
You can access these arguments straight in your ViewModel by using `SavedStateHandle`

```
class AgendaDetailViewModel(
  private val savedStateHandle: SavedStateHandle
) : ViewModel() {
  
  // Fetch single property using key that matches a param name of your route
  val isInEditMode = savedStateHandle[ARG_IS_EDIT_MODE] ?: false
  
  // Fetch single param as a StateFlow
  val isInEditModeFlow: StateFlow<Boolean> = savedStateHandle.getStateFlow(ARG_IS_EDIT_MODE, false)
  
  // Get the entire route object
  val routeArguments = savedStateHandle.toRoute<AgendaDetails>()
}
```